### PR TITLE
node: enable UTs

### PR DESCRIPTION
**Pull request by lmm**
## Description

This PR enables node UTs, which were not enabled, probably because we didn't have any (most of the node tests are system tests and FVs). But at some point we added a Windows CNI config unit test, and a few months ago we added some Windows upgrade tests.

I copied some of the `ut` target from the private version of the Makefile since it makes sense and will streamline merging.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: node/cmd/calico/cni_conf_test.go [hunk 1]
```diff
@@ -23,13 +23,17 @@ import (
 	. "github.com/onsi/gomega"
 )
 
-var _ = Describe("CNI config template tests", func() {
+var _ = Describe("Windows CNI config template tests", func() {
```
Comment (caseydavenport) [inline node/cmd/calico/cni_conf_test.go:26]: huh, I know this isn't added in your PR but I'm not sure why this test file is in the cmd package - it's not testing the cmd package functionality right?

File: node/cmd/calico/cni_conf_test.go [hunk 2]
```diff
@@ -23,13 +23,17 @@ import (
...
 	It("should be valid JSON", func() {
-		f, err := ioutil.ReadFile("../../windows-packaging/TigeraCalico/cni.conf.template")
```
Comment (caseydavenport) [inline node/cmd/calico/cni_conf_test.go:28]: do we need to move a file in order to make this change?

File: node/Makefile [hunk 3]
```diff
@@ -529,7 +538,7 @@ st: $(REMOTE_DEPS) image dist/calicoctl busybox.tar calico-node.tar workload.tar
 # CI/CD
 ###############################################################################
 .PHONY: ci
-ci: static-checks fv image build-windows-archive st
+ci: static-checks ut fv image build-windows-archive st
```
Comment (caseydavenport) [inline node/Makefile:541]: how long do the UTs take to run?

File: node/Makefile [hunk 4]
```diff
@@ -389,8 +389,17 @@ run-k8s-apiserver: stop-k8s-apiserver run-etcd
 stop-k8s-apiserver:
 	@-docker rm -f st-apiserver
 
+GINKGO = ginkgo
+
+UT_PACKAGES_TO_SKIP?=pkg/lifecycle/startup,pkg/allocateip,pkg/status
+.PHONY: ut
+ut: CMD = go mod download && $(GINKGO) -r
```
Comment (caseydavenport) [inline node/Makefile:396]: I don't think we need `go mod download`.

We can also get rid of the layers of variables in use here - let's just do this:

```
ut:
  $(DOCKER_GO_BUILD) sh -c '$(GIT_CONFIG_SSH) ginkgo -r -skipPackage` 
```

File: node/Makefile [hunk 5]
```diff
@@ -389,8 +389,17 @@ run-k8s-apiserver: stop-k8s-apiserver run-etcd
 stop-k8s-apiserver:
 	@-docker rm -f st-apiserver
 
+GINKGO = ginkgo
```
[the remark this answers is not in the document]
Comment (lmm) [reply node/Makefile:392]: Sorry for the late response. I had added this to ease merging but looking at the other repo, I think you're right: I'll remove this.

File: node/Makefile [hunk 6]
```diff
@@ -389,8 +389,17 @@ run-k8s-apiserver: stop-k8s-apiserver run-etcd
...
+
+UT_PACKAGES_TO_SKIP?=pkg/lifecycle/startup,pkg/allocateip,pkg/status
```
[the remark this answers is not in the document]
Comment (lmm) [reply node/Makefile:394]: These are packages with FV tests. I'll add a comment.

File: node/Makefile (hunk 3 above)
Comment (lmm) [reply node/Makefile:541]: They won't add much at all to the pipeline time, it's just a few tests.
```
Ginkgo ran 2 suites in 1.838781692s
Test Suite Passed

real    0m2.230s
user    0m0.078s
sys     0m0.025s
```

File: node/cmd/calico/cni_conf_test.go (hunk 1 above)
Comment (lmm) [reply node/cmd/calico/cni_conf_test.go:26]: Yeah you're right, though I can sort of see why they were added here. Let me find a better place.

File: node/cmd/calico/cni_conf_test.go (hunk 2 above)
Comment (lmm) [reply node/cmd/calico/cni_conf_test.go:28]: The original path is wrong and doesn't exist.

File: node/Makefile (hunk 6 above)
[the remark this answers is not in the document]
Comment (caseydavenport) [reply node/Makefile:394]: What are we defining as an FV vs a UT here?

[the remark this answers is not in the document]
Comment (lmm) [reply node/Makefile:393]: My understanding is that UTs are standalone tests but the FVs require infra to run. We run node UTs in private with the same `UT_PACKAGES_TO_SKIP`.

File: node/cmd/calico/cni_conf_test.go (hunk 1 above)
Comment (lmm) [reply node/cmd/calico/cni_conf_test.go:26]: I've moved them into the a subdir inside `windows-packaging`: 4f9376c1f0d0ae41e5cf2ad21814c3fa219f6190

I checked that the tests are not included in the calico-windows.zip file

File: node/pkg/winupgrade/upgrade_test.go [hunk 7]
```diff
@@ -40,9 +38,6 @@ const (
 var _ = DescribeTable("verifyImagesShareRegistryPath",
 	func(upgradeImage string, nodeImage string, noError bool) {
 		err := verifyImagesSharePathPrefix(upgradeImage, nodeImage)
-		if err != nil {
-			fmt.Println(err)
-		}
 		Expect(err == nil).To(Equal(noError))
```
Comment (caseydavenport) [inline node/pkg/winupgrade/upgrade_test.go:41]: Not your change, but this is a bit bizarre - should be `Expect(err).NotTo(HaveOccurred())`

**Review by caseydavenport — APPROVED** [review]

**Pull request merged**